mailbag/parsers/mbox.py

```diff
@@ -0,0 +1,45 @@
+from mailbag.parser import EmailFormatParser
+from mailbag.emailModel import Email
+import mailbox
+
+
+class Mbox(EmailFormatParser):
+    """Mbox - This concrete class parses mbox file format"""
+    parser_name = 'mbox'
+ 
+    def parse(self, *args, **kwargs):
+        """Parses the mbox file
+        
+        Args:
+            'file' (key) : The file path of mbox file
+        Return:
+            allMails (list) : list of email model objects
+        Example:
+            parse(file="/sample.mbox")
+        """
+        
+        allEmails = []
+        mbox_file = kwargs['file']
+        print("Reading mails from :", mbox_file)        
+        mbox = mailbox.mbox(mbox_file)
+        
+        for key, message in mbox.iteritems():
+
+            try:
+                message
```

Comment:
I don't think just referencing the message would trigger an error - I think you'll need to move all the operations that pull stuff out of the email - lines 33 - 44 - into the try block for it to catch anything. I've been having trouble verifying this because of how permissive mailbox is (I haven't actually managed to trigger the error in question yet), but I'm pretty sure.